From 6f77d9e54afd9fc3c328232dd41549d2e2cc9220 Mon Sep 17 00:00:00 2001
From: Ulrich Hecht <uli@basicengine.org>
Date: Tue, 8 Feb 2022 22:07:21 +0100
Subject: [PATCH] tcc.h: fix interpreter name for x86_64/uclibc

---
 tcc.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tcc.h b/tcc.h
index ef69ef5..05ed160 100644
--- a/tcc.h
+++ b/tcc.h
@@ -312,7 +312,11 @@ extern long double strtold (const char *__nptr, char **__endptr);
 # elif defined(TCC_TARGET_PE)
 #  define CONFIG_TCC_ELFINTERP "-"
 # elif defined(TCC_UCLIBC)
-#  define CONFIG_TCC_ELFINTERP "/lib/ld-uClibc.so.0" /* is there a uClibc for x86_64 ? */
+#  if defined(TCC_TARGET_X86_64)
+#   define CONFIG_TCC_ELFINTERP "/lib/ld64-uClibc.so.0"
+#  else
+#   define CONFIG_TCC_ELFINTERP "/lib/ld-uClibc.so.0"
+#  endif
 # elif defined TCC_TARGET_ARM64
 #  if defined(TCC_MUSL)
 #   define CONFIG_TCC_ELFINTERP "/lib/ld-musl-aarch64.so.1"
-- 
2.20.1

